spec/index.bs

```diff
@@ -762,36 +769,74 @@ enum GPUCompareFunction {
 Resource Binding {#binding}
 ===========================
 
-## GPUPipelineLayout ## {#pipeline-layout}
+## GPUBindGroupLayout ## {#bind-group-layout}
+
+A {{GPUBindGroupLayout}} defines an interface between a set of resources bound in a {{GPUBindGroup}} and their accessibility in shader stages.
 
 <script type=idl>
-interface GPUPipelineLayout : GPUObjectBase {
+interface GPUBindGroupLayout : GPUObjectBase {
 };
 </script>
 
-### Creation ### {#pipeline-layout-creation}
+A {{GPUBindGroupLayout}} object has the following internal slots:
 
-<script type=idl>
-dictionary GPUPipelineLayoutDescriptor : GPUObjectDescriptorBase {
-    required sequence<GPUBindGroupLayout> bindGroupLayouts;
-};
-</script>
+<dl dfn-type=attribute dfn-for="GPUBindGroupLayout">
+	: <dfn>\[[bindingsMap]]</dfn> of a Map of `unsigned long` {{GPUBindGroupLayoutBinding/binding}} keys to <dfn dfn for ="layout binding">layout binding</dfn> values.
+	::
+        The [=layout binding=] objects this {{GPUBindGroupLayout}} describes.
 
-## GPUBindGroupLayout ## {#bind-group-layout}
+</dl>
 
-<script type=idl>
-interface GPUBindGroupLayout : GPUObjectBase {
-};
-</script>
+An internal [=layout binding=] object has the following slots:
+
+<dl dfn-type=attribute dfn-for="layout binding">
+    : <dfn>\[[binding]]</dfn> of type unsigned long.
+    ::
+        The unique identifier for this [=layout binding=] in shader stages.
+
+    : <dfn>\[[visibility]]</dfn> of type {{GPUShaderStageFlags}}.
+    ::
+        The bitset representing from which shader stages this [=layout binding=]'s resource will be accessible.
+
+    : <dfn>\[[type]]</dfn> of type {{GPUBindingType}}.
+    ::
+        The type of shader resource this [=layout binding=] describes.
+
+    : <dfn>\[[hasDynamicOffsets]]</dfn> of type boolean.
+    ::
+        If `true`, indicates that this [=layout binding=] has dynamic offsets.
+
+        Ignored if {{layout binding/[[type]]}} is not {{GPUBindingType/uniform-buffer}}, {{GPUBindingType/storage-buffer}}, or {{GPUBindingType/readonly-storage-buffer}}.
```

Comment:
Why would we want to allow it to be `true` for types that don't have dynamic offsets?
Imagine how you'd have to specify the `set_bind_group()` with this, how you'd define the expected offsets. if we are strict here, then `set_bind_group` becomes trivial to define: one offset is required for each dynamic-offset binding.

Reply:
Hmm, I went with the ignore path to make it a little more flexible. If it's only allowed for certain resources, I'm leaning towards splitting GPUBindGroupLayoutBinding into multiple different types, rather than switching on type.

Reply:
Is there a case where this flexibility would be needed and/or beneficial?

Reply:
Less code to change if you wanted to update the type of a resource binding, I guess? I'm going to make a new PR to see what different types of GPUBindGroupLayoutBinding would look like.